### PR TITLE
[Merged by Bors] - Fix create-reporting-task to support multiple rolegroups

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Fixes bugs found in #452 such as
```
2023-04-10T15:46:25.281021Z ERROR stackable_operator::logging::controller: Failed to reconcile object controller.name="nificluster.nifi.stackable.tech" error=reconciler for object NifiCluster.v1alpha1.nifi.stackable.tech/test-nifi.
default failed error.sources=[failed to apply create ReportingTask job, Kubernetes reported error: ApiError: Job.batch "test-nifi-create-reporting-task-1-18-0" is invalid: spec.template: Invalid value: core.PodTemplateSpec{ObjectMe
ta:v1.ObjectMeta{Name:"test-nifi-create-reporting-task-1-18-0", GenerateName:"" ...  field is immutable: Invalid]
```

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/crd/src/lib.rs [hunk 1]
```diff
@@ -367,15 +367,15 @@ pub struct NifiStorageConfig {
 
 impl NifiCluster {
     /// The name of the role-level load-balanced Kubernetes `Service`
-    pub fn node_role_service_name(&self) -> Option<String> {
-        self.metadata.name.clone()
+    pub fn node_role_service_name(&self) -> String {
+        self.name_unchecked()
```
Comment (maltesander) [inline rust/crd/src/lib.rs:371]: I prefer the name_any to not panic the operator.
```suggestion
        self.name_any()
```

File: rust/operator-binary/src/controller.rs [hunk 2]
```diff
@@ -1107,18 +1104,13 @@ async fn build_node_rolegroup_statefulset(
 fn build_reporting_task_job(
     nifi: &NifiCluster,
     resolved_product_image: &ResolvedProductImage,
-    rolegroup_ref: &RoleGroupRef<NifiCluster>,
     resolved_auth_conf: &ResolvedAuthenticationMethod,
 ) -> Result<Job> {
-    let rolegroup_obj_name = rolegroup_ref.object_name();
-    let namespace: &str = &nifi.namespace().context(ObjectHasNoNamespaceSnafu)?;
+    let nifi_name = nifi.name_unchecked();
```
Comment (maltesander) [inline rust/operator-binary/src/controller.rs:1109]:
```suggestion
    let nifi_name = nifi.name_any();
```

Comment (maltesander) [inline rust/operator-binary/src/controller.rs:1109]:
```suggestion
    let nifi_name = nifi.name_any();
```

File: rust/operator-binary/src/controller.rs [hunk 3]
```diff
@@ -991,7 +985,10 @@ async fn build_node_rolegroup_statefulset(
             ..Volume::default()
         })
         // One volume for the keystore and truststore data configmap
-        .add_volume(build_keystore_volume(KEYSTORE_VOLUME_NAME))
+        .add_volume(build_keystore_volume(
+            KEYSTORE_VOLUME_NAME,
+            &nifi.name_unchecked(),
```
Comment (maltesander) [inline rust/operator-binary/src/controller.rs:990]:
```suggestion
            &nifi.name_any(),
```

**Review by maltesander — APPROVED** [review]
Thanks! LGTM!

**Comment by sbernauer** [comment]
Started https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/nifi-operator-it-custom/31

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**